osu.Game/Database/MemoryCachingComponent.cs

```diff
@@ -39,6 +40,15 @@ protected async Task<TValue> GetAsync([NotNull] TLookup lookup, CancellationToke
             return computed;
         }
 
+        protected void Invalidate(Func<TLookup, bool> invalidationFunction)
```

Comment:
Param needs a better name like `matchKeyPredicate` or something (or xmldoc). I wasn't sure what this was supposed to mean initially.